rosidl_generator_cpp/include/rosidl_generator_cpp/MessageTypeSupport.h

```diff
@@ -17,13 +17,22 @@
 #define __rosidl_generator_cpp__MessageTypeSupport__h__
 
 #include <rosidl_generator_c/message_type_support.h>
+#include <rosidl_generator_c/visibility_control.h>
 
 namespace rosidl_generator_cpp
 {
 
 template<typename T>
+ROSIDL_PUBLIC
 const rosidl_message_type_support_t * get_type_support_handle();
 
 }  // namespace rosidl_generator_cpp
 
+// Tail include the implementation for this function.
+// This include is provided by the implementation and therefore only
+// one version of this header will get included from the path.
+// This header maps the get_type_support_handle function to an implementation
+// specific version of that function.
+#include <rosidl_generator_cpp/get_type_support_handle_impl.hpp>
```

Comment:
Ok, I get the idea I think, but this comment doesn't make sense to me:

> if no implementation is on the include path it will pick up the impl header file from this package giving a really good error using #error

There is no `get_type_support_handle_impl.hpp` in this package, so if user land code (just `rclcpp` for now) includes `rosidl_generator_cpp/get_type_support_handle_impl.hpp` and the include path is not set to one of the implementation's special path, it will not find any header, so there is nothing to put a `#error` in.